proposals/compatibility-mode.md

```diff
@@ -16,24 +16,81 @@ Since WebGPU Compatibility mode is a subset of WebGPU, all valid Compatibility m
+
+EXTENSIBILITY:
+Incremental capabilities between compat and core can be exposed as limits or features which are optional on Compat-defaulting adapters.
+They would be similarly always available and enabled by default on Core-requested adapters
+(including in Core-only implementations).
+
+### Initialization
+
+No IDL changes.
+
+The core spec already allows `featureLevel: "compatibility"`, which is currently ignored.
+When an application passes `"compatibility"`, it indicates that it can handle (and hopefully use) an adapter with capabilities lower than the `"core"` defaults. In this case:
+
+- If a User Agent supporting Compat is running on a Core-*incapable*, it returns a Compat-defaulting Core-incapable adapter (or null).
+- If a User Agent supporting Compat is running on a Core-*capable*, it **may** return a Compat-defaulting Core-capable adapter, or **may** return a Core-defaulting adapter.
+    - ALTERNATIVE: "require" (non-normatively encourage) UAs to go one way or the other - either always emulate compat or always upgrade to core. This is probably not necessary.
+- As already in the current spec, Core-only implementations must still recognize `"compatibility"` but ignore it (automatically upgrade the request) and return a Core-defaulting adapter (or null).
+
+The `GPUAdapter` does not expose whether it is Core-defaulting or Compat-defaulting.
+Most applications shouldn't need to check that compat is enforced, instead requesting the capabilities they want and only using those capabilities.
+Applications that do need to check that compat is enforced (such as tests) should create a device and then check whether the `"webgpu-core"` feature is enabled on the device.
+
+#### Example
+
+```js
+const adapter = await navigator.gpu.requestAdapter()
+const device = await adapter.requestDevice();
 ```
 
-When calling `GPU.RequestAdapter()`, passing `compatibilityMode = true` in the `GPURequestAdapterOptions` will indicate to the User Agent to select the Compatibility subset of WebGPU. Any Devices created from the resulting Adapter on supporting UAs will support only Compatibility mode. Calls to APIs unsupported by Compatibility mode will result in validation errors.
+and
 
-Note that a supporting User Agent may return a `compatibilityMode = true` Adapter which is backed by a fully WebGPU-capable hardware adapter, such as D3D12, Metal or Vulkan, so long as it validates all subsequent API calls made on the Adapter and the objects it vends against the Compatibility subset.
+```js
+const adapter = await navigator.gpu.requestAdapter({ featureLevel: 'compatibility' });
+const device = await adapter.requestDevice({ requiredFeatures: ['webgpu-core'] });
+```
 
-```webidl
-partial interface GPUAdapter {
-    readonly attribute boolean isCompatibilityMode;
+have the same result. Both will succeed if core is available and fail if not.
+
+Using the latter method is a way to upgrade functionality at runtime, but get compat if available, as in:
```

Comment:
Nit: this isn't really the latter method, since the feature is checked after requesting the adapter. I'd just describe the functionality of what follows without referring to the previous example.